File: frontend/src/types/todo.ts

// Data interfaces
export interface Todo {
  id: number
  content: string
  done: boolean
  dueDate?: Date
  description?: string

  belongedListId: number
  createdBy: number
  createdAt: Date
  updatedBy?: number
  updatedAt?: Date
}

export interface TodoList {
  id: number
  name: string
  inbox: boolean

  createdBy: Number
  createdAt: Date
  updatedBy: Number
  updatedAt: Date
}

// Request interfaces
export interface TodoAddReq {
  content: string
  dueDate?: Date
  belongedListId: number
}

export interface TodoUpdateReq {
  id: number
  content: string
  done: boolean
  dueDate?: Date
}

export interface TodoMoveReq {
  id: number
  targetListId: number
}

File: backend/internal/model/response/todo.go
package response

import "time"

type Todo struct {
	ID             uint       `json:"id"`
	Content        string     `json:"content"`
	DueDate        *time.Time `json:"dueDate"`
	Done           bool       `json:"done"`
	Description    *string    `json:"description"`
	BelongedListID uint       `json:"belongedListId"`

	CreatedBy uint       `json:"createdBy"`
	CreatedAt time.Time  `json:"createdAt"`
	UpdatedBy *uint      `json:"updatedBy"`
	UpdatedAt *time.Time `json:"updatedAt"`
}
